### PR TITLE
solved the problem

### PR DESCRIPTION
**Pull request by ravivish**

### CODE REVIEW

File: src/com/snake/and/ladder/Game.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
+package com.snake.and.ladder;
```
Comment (gcnit) [inline src/com/snake/and/ladder/Game.java:1]: This class is doing a lot of things. Should be broken into at least two classes where one of them should control the flow including scanning data.

File: src/com/snake/and/ladder/Game.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class Game {
+
+  private List<Snake> snake;
```
Comment (gcnit) [inline src/com/snake/and/ladder/Game.java:9]: snake, ladder and person also denotes singular things whereas a list denotes plurality. These should be named snakes, ladders, persons.

File: .gitignore [hunk 3]
```diff
@@ -1,2 +1,4 @@
 .idea
-.DS_Store
\ No newline at end of file
+.DS_Store
+*.iml
+.target
```
Comment (gcnit) [inline .gitignore:4]: put out folder as well to get your code reviewed

File: src/com/company/Board.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
+package com.company;
+
+import java.util.Arrays;
+
+public class Board implements IBoard {
+
+  private final int Row;
```
Comment (gcnit) [inline src/com/company/Board.java:7]: property name should be lowercase. You already know that so why starting with capital letter?

File: src/com/company/Board.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  private final int Column;
+  String[][] board;
+
+  public Board(int row, int column) {
+    Row = row;
+    Column = column;
+    board = new String[Row][Column];
+    SetBoard(board);
```
Comment (gcnit) [inline src/com/company/Board.java:15]: function name should be lowercase. You already know that so why starting with capital letter?

File: src/com/company/Board.java [hunk 6]
```diff
@@ -0,0 +1,100 @@
+package com.company;
+
+import java.util.Arrays;
+
+public class Board implements IBoard {
```
Comment (gcnit) [inline src/com/company/Board.java:5]: It was specifically mentioned to not use I prefix for interface. Make sure you are following the practices mentioned in the sessions to get your code reviewed. There is no point in mentioning the same things here that should have ideally been followed after the session. The code review is mostly for design and not these practices.

File: src/com/company/Board.java [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+  @Override
+  public boolean checkHorizontalWin(Players player) {
+    for (int i = 0; i < Row; i++) {
+      int pieceCount = 0;
+      for (int j = 0; j < Column; j++) {
+        if (board[i][j].equals(player.getPiece())) {
+          ++pieceCount;
+        }
+      }
+      if (pieceCount == 3) {
```
Comment (gcnit) [inline src/com/company/Board.java:60]: magic number

File: src/com/company/Board.java [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+    }
+    return false;
+  }
+
+  @Override
+  public boolean checkDiagonalWin(Players player) {
+    int pieceCount = 0;
+    for (int i = 0; i < Row; i++) {
+      for (int j = 0; j < Column; j++) {
+        if (i == j && board[j][i].equals(player.getPiece())) {
```
Comment (gcnit) [inline src/com/company/Board.java:88]: Have you tested this logic?
It is wrong.

You are doing 3*3 loops. You will get count > 3 for diagonal case with this logic.

File: src/com/company/GameService.java [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+  IBoard iBoard;
+  int Row = 3;
+  int Column = 3;
+
+  public GameService() {
+    this.iBoard = new Board(Row, Column);
+  }
+
+  public boolean nextMove(int row, int column, Players player) {
```
Comment (gcnit) [inline src/com/company/GameService.java:13]: this is not a verb

File: src/com/company/Players.java [hunk 10]
```diff
@@ -0,0 +1,21 @@
+package com.company;
+
+public class Players implements IPlayers {
```
Comment (gcnit) [inline src/com/company/Players.java:3]: Why is this called Players when it denotes a single player?

File: src/com/company/GameService.java [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+
+    iBoard.setPiece(row, column, player);
+    iBoard.printBoard();
+    if (iBoard.checkDiagonalWin(player)) {
+      System.out.println(player.getName() + " won the game");
+    }
+    if (iBoard.checkHorizontalWin(player)) {
```
Comment (gcnit) [inline src/com/company/GameService.java:20]: Always use else. Your code will print the same thing twice in certain cases based on the current code.

File: src/com/company/Board.java (hunk 4 above)
Comment (ravivish) [reply src/com/company/Board.java:7]: ok, will fix that

File: src/com/company/Board.java (hunk 5 above)
Comment (ravivish) [reply src/com/company/Board.java:15]: ok, will fix that

File: src/com/company/Board.java (hunk 6 above)
Comment (ravivish) [reply src/com/company/Board.java:5]: my company follow that, I won't do this in next problem

File: src/com/company/Board.java (hunk 7 above)
Comment (ravivish) [reply src/com/company/Board.java:60]: should use variable for that right ?

File: src/com/company/GameService.java (hunk 11 above)
Comment (ravivish) [reply src/com/company/GameService.java:20]: ok

File: src/com/company/Players.java (hunk 10 above)
Comment (ravivish) [reply src/com/company/Players.java:3]: ok, It should be Player

File: src/com/company/Board.java (hunk 8 above)
Comment (ravivish) [reply src/com/company/Board.java:88]: yes it was working with sample testcase, but I forgot to add the right diagonal logic

File: src/com/company/GameService.java (hunk 9 above)
Comment (ravivish) [reply src/com/company/GameService.java:13]: ok, will fix that

File: .gitignore (hunk 3 above)
Comment (ravivish) [reply .gitignore:4]: ok

File: src/com/company/Board.java (hunk 7 above)
Comment (gcnit) [reply src/com/company/Board.java:60]: Constant

**Pull request closed without merging**